config/src/main/java/io/confluent/common/config/ConfigDef.java

```diff
@@ -361,10 +366,14 @@ else if (value instanceof String)
           if (value instanceof Class) {
             return (Class<?>) value;
           } else if (value instanceof String) {
-            if (Thread.currentThread().getContextClassLoader() != null) {
-              return Class.forName(trimmed, true, Thread.currentThread().getContextClassLoader());
-            } else {
+            try {
               return Class.forName(trimmed);
+            } catch (ClassNotFoundException e) {
+              log.error("Could not find class " + trimmed, e);
+              if (Thread.currentThread().getContextClassLoader() != null) {
+                log.error("Reattempting with context class loader " + trimmed, e);
+                return Class.forName(trimmed, true, Thread.currentThread().getContextClassLoader());
+              }
```

Comment:
@mageshn, @kkonstantine does this change make sense to you? It use the default classloader (the one which originally created this class), before using the ContextClassLoader to instantiate classes. This is going to impact how Connect's plugin classloader works. If this change is incorrect, we need to rethink how we did https://github.com/confluentinc/schema-registry/commit/d8fd3bf6feaeb49433e8474e284b6cc0aabe713c (or how AvroConverter's dependencies are laid out in CP).